xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -982,28 +975,31 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   // Figure out the screen size. (default to main screen)
   const CGDirectDisplayID display_id = GetDisplayID(screenIdx);
 
-  // find mode that matches the desired size, refreshrate
-  // non interlaced, nonstretched, safe for hardware
-  dispMode = GetMode(width, height, refreshrate, screenIdx);
+  // find mode that matches the desired size, refreshrate non interlaced, nonstretched, safe for hardware
 
-  //not found - fallback to bestemdeforparameters
+  // try to find an exact match first (by using the unique ids we assign to resolution infos)
+  dispMode = CreateModeById(res.strId, screenIdx);
   if (!dispMode)
   {
-    dispMode = BestMatchForMode(display_id, 32, width, height);
+    dispMode =
+        CreateMode(res.iWidth, res.iHeight, static_cast<double>(res.fRefreshRate), screenIdx);
+  }
+
+  // not found - fallback to bestmodeforparameters
+  if (!dispMode)
+  {
+    dispMode = BestMatchForMode(display_id, 32, res.iWidth, res.iHeight);
```

Comment:
and now you have a mix of retained dispMode from above calls and non-retained from this and next BestMatchForMode call - one more reason to change BestMatchForMode to return retained value